src/utils/messaging-outbox/src/services/outbox_transactional_processor.rs

```diff
@@ -77,32 +88,47 @@ impl OutboxTransactionalProcessor {
 
     #[tracing::instrument(level = "debug", skip_all)]
     pub async fn run(&self) -> Result<(), InternalError> {
-        // Trace current routes
-        self.debug_message_routes();
-
-        // Make sure consumption records represent the routes
-        self.init_consumption_records().await?;
-
-        // Main relay loop
-        loop {
-            self.run_relay_iteration().await?;
+        self.run_with_mode(RunMode::MainRelayLoop).await
+    }
 
-            tracing::debug!("Awaiting next iteration");
-            tokio::time::sleep(self.config.awaiting_step.to_std().unwrap()).await;
-        }
+    #[tracing::instrument(level = "debug", skip_all)]
+    pub async fn run_while_has_messages(&self) -> Result<(), InternalError> {
+        self.run_with_mode(RunMode::WhileHasMessages).await
     }
 
     // To be used by tests only!
     #[tracing::instrument(level = "debug", skip_all)]
     pub async fn run_single_iteration_only(&self) -> Result<(), InternalError> {
+        self.run_with_mode(RunMode::SingleIterationOnly).await
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn run_with_mode(&self, mode: RunMode) -> Result<(), InternalError> {
         // Trace current routes
         self.debug_message_routes();
 
         // Make sure consumption records represent the routes
         self.init_consumption_records().await?;
```

Comment:
I think I've moved these lines to new `pre_run` part. Careful when merging.

Reply:
Thanks for the comment, I've taken into account